config/application.rb

```diff
@@ -153,7 +153,20 @@ class Application < Rails::Application
 
     # enables in memory cache store with ttl
     # config.cache_store = TimedCachedStore.new
-    config.cache_store = :file_store, Rails.root.join("tmp", "cache")
+    rails_cache_settings = SETTINGS[:rails_cache_store]
+    if rails_cache_settings && rails_cache_settings[:type] == 'redis'
+      args = [:redis_cache_store]
+      if ENV['RAILS_CACHE_STORE_HOSTS'].present?
```

Comment:
can you swap the condition that if redis env is defined the cache is also set? my aim is to expose it via docker env without the settings file..

Reply:
@ohadlevy: Sure, like this?

Reply:
I have my concerns about adding environment variables for configuration. Not against it in general but I think we can end up with a very mixed and undocumented set of env vars.

Can't we read env vars in the settings layer and then just use the `SETTINGS` class in here? It at least keeps a clean layer in between.

Reply:
yes, just wondering if the name of the variable makes sense, its redis specific vs rails cache store in general?also, do we really assume we have multiple cache servers? (I can see multiple redis with persisitent data, less for cache) so all in all in my opinion, I think ```REDIS_URL``` is easier and perhaps can add multi host with ```REDIS_URL='redis://one redis://two'```...

Reply:
> I have my concerns about adding environment variables for configuration. Not against it in general but I think we can end up with a very mixed and undocumented set of env vars.
> 
> Can't we read env vars in the settings layer and then just use the `SETTINGS` class in here? It at least keeps a clean layer in between.

like https://github.com/theforeman/foreman/pull/6811#discussion_r289146370?

Reply:
I think we should first decide if we want to configure a default redis connection that will be used for every use case or allow a user to configure separate instances. This will answer if we need to prefix the variable with `RAILS_CACHE_STORE_` or not.
I think we should definitely allow setting multiple hosts.

> Can't we read env vars in the settings layer and then just use the SETTINGS class in here?
@ekohl: That would be great, but we'd need to cast these variables. SETTINGS basically just loads a YAML-file, see https://github.com/theforeman/foreman/blob/develop/config/boot_settings.rb. We can, of course, hardcode all supported settings there. Maybe that's not such a bad idea.

Reply:
I'd be in favor of merging this without environment variables (since Redis is useful on its own) and then opening the bikeshedding discussion.